### PR TITLE
docs: Signal::recv is cancel-safe

### PR DESCRIPTION
**Pull request by matklad**
I think this is the case, so let's document it!

### CODE REVIEW

**Comment by Darksonn** [comment]
That list is not exhaustive. Is it documented on the documentation for the method itself?

**Comment by matklad** [comment]
Added method level docs!

I guess, we can keep it *only* as method docs, but it might be helpful to see it in the main list as well -- signals is something you are fairly likely to use with select for cancellation

**Comment by ipetkov** [comment]
Just confirming that yes, tokio's signal APIs are cancel safe. 

As far as the docs are concerned I'm happy to have them both in the method and the general list

**Review by ipetkov — APPROVED** [review]

File: tokio/src/signal/unix.rs [hunk 1]
```diff
@@ -407,6 +407,12 @@ impl Signal {
     ///
     /// `None` is returned if no more events can be received by this stream.
     ///
+    /// # Cancel safety
+    ///
+    /// This method is cancel safe. If you use it as the event in a
+    /// [`tokio::select!`](crate::select) statement and some other branch
+    /// completes first, then it is guaranteed that signal is not lost.
```
Comment (Darksonn) [inline tokio/src/signal/unix.rs:414]:
```suggestion
    /// completes first, then it is guaranteed that no signal is lost.
```

**Pull request merged**